src/app/core/services/geolocation.service.ts

```diff
@@ -26,7 +26,17 @@ export class GeolocationService {
 
   getPosition(useCache = true): Observable<GeolocationPosition> {
     const cache = (this.isCachedPositionValid() && useCache);
-    const position$ = (cache) ? of(this.cachedPosition) : from(Geolocation.getCurrentPosition(this.defaultGeolocationOptions));
+
+    const getPosition = (): Promise<GeolocationPosition> =>
+      Geolocation
+        .getCurrentPosition(this.defaultGeolocationOptions)
+        .catch<GeolocationPosition>((error) => (error.message as string).includes('0')
```

Comment:
The meaning of `(error.message as string).includes('0')` is unclear. It's confusing to people who read the code but don't know what error you are catching, so it's better to rewrite the code in a way that it's easier to understand. Or if it's difficult to be expressive enough in code, write some comments to explain it.

And it doesn't seem necessary to create another function if all you want to do is catching errors. You can catch error using RxJS pipe operator `catchError` chained after `position$`, for example

```
return position$
  .pipe(
    catchError(error => of(undefined)),
```

The above example will  send `undefined` to be handled by the next operator in the pipe if position$ emits error.